### PR TITLE
feat: --skip-validation allows mismatched Augur versions

### LINKED ISSUE #901: ENH: Add flag that ignores version mismatch checks in `export`
My augur workflow contains runs a custom version of refine based on Augur 14.0.0 with FFT treetime PR patches on top.

However, the rest uses Augur 15.0.0. Now `export` complains that `branch lenghts` was produced by version 14 and says the versions are incompatible. That's actually not true, export would still work, `augur` is just being overly careful.

We introduced an option to skip output json validation in v14, I think it could be useful for situations like mine to have a flag `--ignore-version-mismatch` in `export`.

It's good to have that warning on by default but it'd be nice to be able to override.

### PR DESCRIPTION
**Pull request by corneliusroemer**
### Description of proposed changes
Currently, files produced by Augur contain the version of the Augur instance used to produce the data file.

When there's a major version mismatch between the version a file was produced with and the version of Augur `export`, Augur throws an error complaining about version mismatch.

While it's a good thing to warn the user, there are situations where one may want to take the risk and allow Augur export to use output from different versions.

This PR extends the meaning of the `--skip-validation` argument to also not complain about version mismatch in input files.

`--skip-validation` was introduced in https://github.com/nextstrain/augur/pull/865

### Related issue(s)
Fixes #901
Related to #865 

### Testing
I added integration tests

### CODE REVIEW

File: tests/functional/export_v2.t [hunk 1]
```diff
@@ -65,3 +65,25 @@ Export with auspice config JSON with an extensions block
...
+
+# export_v2/location_node-data2.json was produced by a mismatched version
+# Running without --skip-validation should result in an error
+# Message printed: "Validation of auspice_config3.json failed."
+  $ ${AUGUR} export v2 \
+  >   --tree export_v2/tree.nwk \
+  >   --node-data export_v2/div_node-data.json export_v2/location_node-data2.json \
+  >   --auspice-config export_v2/auspice_config3.json \
+  >   --output "$TMP/dataset4.json" &>/dev/null
+  [1]
```
Comment (huddlej) [inline tests/functional/export_v2.t:77]: Instead of commenting what the message should be and piping stderr to null, I would allow stderr to print and put the expected message in the output of the command above `[1]`. This way the Cram test documents and tests the expected behavior.

File: tests/functional/export_v2.t [hunk 2]
```diff
@@ -65,3 +65,25 @@ Export with auspice config JSON with an extensions block
...
+
+# Skipping validation gives us the same results as `location_node-data.json`
+  $ ${AUGUR} export v2 \
+  >   --tree export_v2/tree.nwk \
+  >   --node-data export_v2/div_node-data.json export_v2/location_node-data2.json \
+  >   --auspice-config export_v2/auspice_config2.json \
+  >   --output "$TMP/dataset5.json" \
+  >   --skip-validation &>/dev/null
```
Comment (huddlej) [inline tests/functional/export_v2.t:85]: Same as above comment, I would not pipe stderr to null. If `--skip-validation` works, we shouldn't get any error messages and the exit code should be 0.

File: augur/util_support/node_data_file.py [hunk 3]
```diff
@@ -9,8 +9,9 @@
 
 
 class NodeDataFile:
-    def __init__(self, fname):
+    def __init__(self, fname, skip_validation=False):
```
Comment (huddlej) [inline augur/util_support/node_data_file.py:12]: I agree it's annoying to pass around this variable through multiple classes. This isn't ideal OO design and an OO implementation is even what we need here. Your implementation here seems fine, in the current context. We should leave to another PR to reimplement the node data reader as a function again. 😄

**Review by huddlej — COMMENTED** [review]
Thank you, @corneliusroemer! I had a couple minor comments about the tests, but otherwise good to merge. (Thank you for adding tests!)

File: augur/util_support/node_data_file.py (hunk 3 above)
Comment (jameshadfield) [reply augur/util_support/node_data_file.py:12]: Yes!

**Comment by jameshadfield** [comment]
I really needed this today so I took it upon myself to implement John's PR suggestions & also rebased it onto current master (99 commits ahead!). I think this should be merged 🙏

File: tests/functional/export_v2.t (hunk 1 above)
Comment (jameshadfield) [reply tests/functional/export_v2.t]: Addressed in [404595d](https://github.com/nextstrain/augur/pull/902/commits/404595d60759a46f3294154ec37211d5ea907feb). A bit tricky as we print the current augur version so need to use a regex on the output.

File: tests/functional/export_v2.t (hunk 2 above)
Comment (jameshadfield) [reply tests/functional/export_v2.t]: Addressed in [404595d](https://github.com/nextstrain/augur/pull/902/commits/404595d60759a46f3294154ec37211d5ea907feb) -- the output is now detailed in all it's beauty 😂

**Comment by jameshadfield** [comment]
Test failure is the same as seen on another PR: https://github.com/nextstrain/augur/pull/961#issuecomment-1143131103 and seems unrelated to this PR 😕

**Comment by victorlin** [comment]
[This was bumped in Slack](https://bedfordlab.slack.com/archives/C01LCTT7JNN/p1659016554564449). Force-pushed a rebase on latest `master`, then I'll merge once tests pass.
